include/net/lwm2m.h

```diff
@@ -1062,6 +1062,22 @@ int lwm2m_engine_update_service_period(k_work_handler_t service, uint32_t period
  */
 int lwm2m_update_device_service_period(uint32_t period_ms);
 
+/**
+ * @brief Check whether a path is observed
+ *
+ * @param[in] pathstr LwM2M path string to check, e.g. "3/0/1"
+ *                    Note: level 4 (resource-instance) is treated the exact same
+ *                    as level 3 (resource) since there exist no resource instance
+ *                    specific observations in LwM2M.
```

Comment:
I'd mention here that this only applies for LwM2M 1.0 - we're currently working on 1.1 support, where RI observations are allowed. Eventually, the function should work in both cases.

Reply:
Actually I think the function could just check the resource instance right now and work as expected when resource instance observations are created.

Reply:
Sounds good!